pyhms/scipy_style.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass
+from typing import Callable
+
+import numpy as np
+from leap_ec.problem import FunctionProblem
+
+from .config import CMALevelConfig, EALevelConfig, Options, TreeConfig
+from .demes.single_pop_eas.sea import SEA
+from .logging_ import LoggingLevel
+from .problem import EvalCutoffProblem
+from .sprout import get_NBC_sprout
+from .stop_conditions import (
+    DontStop,
+    GlobalStopCondition,
+    MetaepochLimit,
+    SingularProblemEvalLimitReached,
+    UniversalStopCondition,
+)
+from .tree import DemeTree
+
+DEFAULT_MAX_FUN = 10000
+
+
+@dataclass
+class OptimizeResult:
+    x: np.ndarray
+    nfev: int
+    fun: float
+    nit: int
+
+
+def parse_log_level(log_level: str | LoggingLevel | None) -> LoggingLevel:
+    if log_level is None:
+        return LoggingLevel.WARNING
+    if isinstance(log_level, LoggingLevel):
+        return log_level
+    levels = [level.value for level in LoggingLevel]
+    if log_level.lower() not in levels:
+        raise ValueError(f"Invalid log level: {log_level}, available levels: {levels}")
+    return LoggingLevel[log_level.upper()]
+
+
+def minimize(
+    fun: Callable[[np.ndarray], float],
+    bounds: np.ndarray,
+    maxfun: int | None = None,
+    maxiter: int | None = None,
+    seed: int | None = None,
+    log_level: str | LoggingLevel | None = LoggingLevel.WARNING,
+) -> OptimizeResult:
+    # If neither maxfun nor maxiter are specified, default to maxfun=10000.
+    if maxfun is None and maxiter is None:
+        maxfun = DEFAULT_MAX_FUN
+    function_problem = FunctionProblem(fun, maximize=False)
+    if maxfun is not None:
+        function_problem = EvalCutoffProblem(function_problem, eval_cutoff=maxfun)
+    gsc: GlobalStopCondition | UniversalStopCondition = (
+        SingularProblemEvalLimitReached(maxfun) if maxfun is not None else MetaepochLimit(maxiter)
+    )
+    level_config = [
+        EALevelConfig(
+            ea_class=SEA,
+            generations=1,
```

Comment:
If we want the users to default to this interface, then we should probably give them an option to set these parameters or come up with some default calculators. Maybe an HMS_options class or passing through **kwargs?

Reply:
I implemented default calculators.